kernel/safe_typing.ml

```diff
@@ -675,24 +677,22 @@ let inline_private_constants_in_definition_entry env ce =
 let inline_private_constants_in_constr env body side_eff =
   pi1 (inline_side_effects env body side_eff)
 
-let rec is_nth_suffix n l suf =
-  if Int.equal n 0 then l == suf
-  else match l with
-  | [] -> false
-  | _ :: l -> is_nth_suffix (pred n) l suf
+let is_suffix l suf = match l with
```

Comment:
This part is fishy, isn't it the case that doing things one by one the mb gets reallocated and == missing it?
I don't recall exactly the old code, but this looks fishy.

Reply:
Nope, the only difference with the old code is that you'd have to check more physical equalities. But anyways, this code was very rarely triggered with `n` not being one, as for this to occur you need to generate rewrite schemes on mutual inductive types.